lib/nrf_modem_lib/nrf_modem_lib_trace.c

```diff
@@ -8,16 +8,30 @@
 #include <kernel.h>
 #include <zephyr.h>
 #include <modem/nrf_modem_lib_trace.h>
+#include <nrf_errno.h>
+#include <nrf_modem.h>
 #include <nrf_modem_at.h>
+#include <logging/log.h>
 #ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART
 #include <nrfx_uarte.h>
 #endif
 #ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT
 #include <SEGGER_RTT.h>
 #endif
 
+LOG_MODULE_REGISTER(nrf_modem_lib_trace, CONFIG_NRF_MODEM_LIB_LOG_LEVEL);
+
+static K_HEAP_DEFINE(trace_heap, CONFIG_NRF_MODEM_LIB_TRACE_HEAP_SIZE);
```

Comment:
Eventually, we want the trace heap inside the os layer, so it could be good to have it in the os layer right from the start. I think passing the trace heap to the trace api on init could be a good solution.

Reply:
Moved the trace heap to the os layer now.